### PR TITLE
Adds initial pass at tmux compatibility

### PR DESCRIPTION
**Pull request by iamkirkbater**
### What type of PR is this?

Feature/Cleanup

### What this PR does / Why we need it?

Reconfigures the launcher to work with Tmux, MacOS default Terminal and adds docs on how to use this with iTerm2 on MacOS. Should contain no breaking changes for existing users.

### Special notes for your reviewer

Will need to be tested with a linux machine. Specifically if there are existing configurations I want to make sure they're not breaking.
~Also still needs some docs updates.~

### Pre-checks (if applicable)

- [x] Ran unit tests locally against the changes
- [x] Included documentation changes with PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,3 +81,85 @@ ignoredusers:
...
+## Automatic Login Configuration
+
+To enable automatic login directly from SREPD you will need to configure the `terminal` and `cluster_login_command` settings in the srepd config file.
+
+### Linux
+A typical linux configuration to launch a new terminal window may look something like what follows. Be sure to change to your preferred terminal or preferred ops environment (ocm-container, ocm-backplane session, osdctl session, etc)
+
+```yaml
+terminal: gnome-terminal
+cluster_login_command: ocm-container -C
```
Comment (clcollins) [inline README.md:94]:
```suggestion
cluster_login_command: ocm backplane login
```

File: pkg/launcher/launcher.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+func (l *ClusterLauncher) BuildLoginCommand(cluster string) ([]string, error) {
+	command := []string{}
+
+	// Handle the Terminal command
+	// The first arg should not be something replaceable, as checked in the
+	// validate function
+	command = append(command, l.terminal[0])
+	command = append(command, replaceVars(l.terminal[1:], cluster)...)
+
+	needsClusterIDAppended := true
```
Comment (clcollins) [inline pkg/launcher/launcher.go:62]: Maybe we just document %%CLUSTER_ID%% and say it needs to be in the command they specify?  We can set `ocm backplane login %%CLUSTER_ID%%` as a default so it's more clear.  Then we don't have to handle both cases.

File: pkg/launcher/launcher.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+			needsClusterIDAppended = false
+		}
+	}
+	loginCmd := replaceVars(l.clusterLoginCommand, cluster)
+
+	if needsClusterIDAppended {
+		loginCmd = append(loginCmd, cluster)
+	}
+
+	if l.settings.collapseLoginCommand {
```
Comment (clcollins) [inline pkg/launcher/launcher.go:74]: I think we can just handle this automatically and not require the user to specify that it needs to be collapsed, or need to know anything about golang or the code to use SREPD.

I'm pretty sure I struggled with something similar in ocm-container golang, but found a solution.  I'll take a look at how I did that.

**Review by clcollins — COMMENTED** [review]
This is great!  The templating is a great idea, and I think that'll give us a lot of flexibility in the future!

I think the collapse_login_command can be removed - I'll look at how I was handling that in ocm-container golang.  I'm pretty sure I was able to split the merged strings and then map them back down.  

WDYT about just making the %%CLUSTER_ID%% a standard in the config, and not deciding to just append it to any argument.  If a default is there for reference, and we document it, I think it would simplify the code and make it easier for new users to onboard.

File: pkg/launcher/launcher.go (hunk 2 above)
Comment (iamkirkbater) [reply pkg/launcher/launcher.go:62]: I'm also +1 to this - but I didn't want to force everyone who's already installed this to have to reconfigure. I'll leave the decision up to you :D

File: README.md (hunk 1 above)
Comment (iamkirkbater) [reply README.md]: Sure! If I had any experience with any of the other session commands I would have added them in here. I hope that as people onboard who have those other workflows that we can start getting them into the docs as well :D

**Review by clcollins — APPROVED** [review]

**Pull request merged**